src/withStyles.jsx

```diff
@@ -57,11 +59,16 @@ export function withStyles(
       }
 
       maybeCreateStyles() {
+        const registeredTheme = ThemedStyleSheet.get();
+        const recreateStyles = !createStylesOnce && (currentTheme !== registeredTheme);
+        currentTheme = registeredTheme;
```

Comment:
I agree. I had that thought too. I figured that `ThemedStyleSheet` shouldn't define the concept of changed this way necessarily (is it when a theme is registered or when the theme changes with `!==`), so I left it out of there. I'm not sure though, I can see it either way.

Reply:
Actually this may be cleaner ^

Reply:
Hmm we'd have to have a way to prevent future recreates after this as well, e.g. once we've built the styleSheet with createRTL/createLTR we could update the themeChanged variable again (trickily, we'd have to ... have a flag for each direction, no?)

Reply:
Wait no, because if the theme changes only once, then styles would be recreated every time.